### DISCUSSION TITLE
Tasmota deep sleep 3.3v

### DISCUSSION THREAD

**Discussion by abhitheawesomecoder** [arendst/Tasmota#16025] [Ideas]
In deepsleep mode 3.3v gpio is available on wemos D1 mini i.e. it's on

Is there a way to keep it off in deepsleep mode and turn it on only when device wakes up?

**Comment by barbudor** [comment]
How would you turn it back on from an unpowered device?
Nothing impossible if you build an additional circuit that will do that but that additional circuit needs to remain on.

I don't see the point but whatever, mini-D1 is not the good hardware to use deepsleep (at least not without some modification)
See this for some info https://youtu.be/rfPwOtoGO4E

**Reply by abhitheawesomecoder** [reply]
> How would you turn it back on from an unpowered device? Nothing impossible if you build an additional circuit that will do that but that additional circuit needs to remain on.
> 
> I don't see the point but whatever, mini-D1 is not the good hardware to use deepsleep (at least not without some modification) See this for some info https://youtu.be/rfPwOtoGO4E

Thanks for prompt reply but what I mean is 3.3v gpio pin is active in deepsleep mode

**Reply by barbudor** [reply]
Yes because it's not dépendant of the ESP.
If you want to control the power applied to some external sensors to turn them off for example, you can power them through a mosfet.
My preference is to make a high side switch with a P-channel mosfet with gate controlled by a GPIO which which has a pull-up (gpio 0 or 2, or any other but need to add an external pull-up resistor. In Tasmota define the GPIO as OUTPUT_LOW
When the esp is inactive the pull-up will get a HI on the gate and the mosfet will be off. When tasmota runs, it will drive the GPIO low, making the mosfet on.

**Reply by abhitheawesomecoder** [reply]
> Yes because it's not dépendant of the ESP. If you want to control the power applied to some external sensors to turn them off for example, you can power them through a mosfet. My preference is to make a high side switch with a P-channel mosfet with gate controlled by a GPIO which which has a pull-up (gpio 0 or 2, or any other but need to add an external pull-up resistor. In Tasmota define the GPIO as OUTPUT_LOW When the esp is inactive the pull-up will get a HI on the gate and the mosfet will be off. When tasmota runs, it will drive the GPIO low, making the mosfet on.

Can I simply use BC547 Transistor?

**Reply by sfromis** [reply]
If your device needs very little power, like many sensors, you can also power it directly from a gpio without adding a transistor, configuring the pin as `Output Hi`.

**Reply by barbudor** [reply]
You will loose ~0.5v of Vce, and a bc547 is NPN, not suitable for high side switch.

Or follow sfromis' advice if your sensor need less than  a few mA

**Reply by sfromis** [reply]
IIRC, the gpio pins are safe for 10 mA.
